src/frontend/src/repositories/anchorMetadata.test.ts

```diff
@@ -0,0 +1,223 @@
+import { MetadataMapV2 } from "$generated/internet_identity_types";
+import {
+  AnchorMetadata,
+  AnchorMetadataRepository,
+  RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS,
+} from "./anchorMetadata";
+
+const recoveryPageShownTimestampMillis = 1234567890;
+const mockRawMetadata: MetadataMapV2 = [
+  [
+    RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS,
+    { String: String(recoveryPageShownTimestampMillis) },
+  ],
+];
+const mockAnchorMetadata: AnchorMetadata = {
+  recoveryPageShownTimestampMillis,
+};
+
+// Used to await that the getter has resolved.
+let getterResponse: MetadataMapV2 | null | undefined = null;
+const getterMockSuccess = vi.fn().mockImplementation(async () => {
+  // The `await` is necessary to make sure that the `getterResponse` is set before the test continues.
+  getterResponse = await mockRawMetadata;
+  return mockRawMetadata;
+});
+const getterMockError = vi.fn().mockImplementation(async () => {
+  // The `await` is necessary to make sure that the `getterResponse` is set before the test continues.
+  getterResponse = await null;
```

Comment:
Sorry, I'm missing some context here. Why does the `await` change anything at all? Shouldn't that instantly resolve anyway?

Reply:
For some reason it wasn't waiting before. But I tried again and it was working now. Removed.